### DISCUSSION TITLE
Regarding the open-source license of ONNX models

### DISCUSSION THREAD

**Discussion by dpnotnull** [infiniflow/ragflow#2257] [Q&A] [answered]
Apologies for the doubt, but from what I've seen on Netron, the layout model and tsr model used by DeepDoc are under the AGPL-3.0 license.

Does this mean that I cannot freely use DeepDoc in my company's commercial projects?

Looking forward to your response, many thanks🙏

**Comment by KevinHuSh** [comment]
Which parts are under  AGPL-3.0?

**Comment by dpnotnull** [comment]
Dear Team:


Many thanks for your reply, and I apologize for the late email response.






When I check the metadata of the .onnx models , found that the tsr.onnx and layout.xxx are based on AGPL-3.0 license:



$ onnx inspect -m ragflow/ragflow/rag/res/deepdoc/tsr.onnx&nbsp;

Meta information

--------------------------------------------------------------------------------

&nbsp; IR Version: 8

&nbsp; Opset Import: [version: 17

]

&nbsp; Producer name: pytorch

&nbsp; Producer version: 2.2.0

&nbsp; Domain:&nbsp;

&nbsp; Doc string:&nbsp;

&nbsp; meta.{} = {} description Ultralytics best model trained on PubTables-1M-yolo-datasets/data.yaml

&nbsp; meta.{} = {} author Ultralytics

&nbsp; meta.{} = {} license AGPL-3.0 https://ultralytics.com/license

&nbsp; meta.{} = {} date 2024-03-25T04:36:35.872713

&nbsp; meta.{} = {} version 8.1.18

&nbsp; meta.{} = {} stride 32

&nbsp; meta.{} = {} task detect

&nbsp; meta.{} = {} batch 1

&nbsp; meta.{} = {} imgsz [640, 640]

&nbsp; meta.{} = {} names {0: 'table', 1: 'table column', 2: 'table row', 3: 'table column header', 4: 'table projected row header', 5: 'table spanning cell', 6: 'table grid cell'}



onnx inspect -m ragflow/ragflow/rag/res/deepdoc/layout.onnx&nbsp;

Meta information

--------------------------------------------------------------------------------

&nbsp; IR Version: 8

&nbsp; Opset Import: [version: 17

]

&nbsp; Producer name: pytorch

&nbsp; Producer version: 2.2.0

&nbsp; Domain:&nbsp;

&nbsp; Doc string:&nbsp;

&nbsp; meta.{} = {} description Ultralytics best model trained on /yifan/CDLA-yolo/data.yaml

&nbsp; meta.{} = {} author Ultralytics

&nbsp; meta.{} = {} license AGPL-3.0 https://ultralytics.com/license

&nbsp; meta.{} = {} date 2024-02-29T01:29:38.132690

&nbsp; meta.{} = {} version 8.1.18

&nbsp; meta.{} = {} stride 32

&nbsp; meta.{} = {} task detect

&nbsp; meta.{} = {} batch 1

&nbsp; meta.{} = {} imgsz [640, 640]

&nbsp; meta.{} = {} names {0: 'none', 1: 'text', 2: 'title', 3: 'figure', 4: 'figure_caption', 5: 'table', 6: 'table_caption', 7: 'header', 8: 'footer', 9: 'reference', 10: 'equation'}


Does this mean that in our company's commercial projects, at least we cannot use these two types of models freely?



Would you please help to check this?


Thanks again🙂


BR,
Yunpeng




                       
Original
                       
                     

From:"Kevin Hu"< ***@***.*** &gt;;

Date:2024/9/11 9:40

To:"infiniflow/ragflow"< ***@***.*** &gt;;

CC:"dpnotnull"< ***@***.*** &gt;;"Author"< ***@***.*** &gt;;

Subject:Re: [infiniflow/ragflow] Regarding the open-source license of ONNXmodels (Discussion #2257)




 
Which parts are under  AGPL-3.0?

**Reply by KevinHuSh** [reply]
It's apache-2.0. We're gona fix that.

**Reply by dpnotnull** [reply]
Thank you for your response. I appreciate your feedback. 

However, I would like to politely inquire about the timeline for addressing this issue. 🙏🙏

Specifically, could you provide an estimate of when the OCR-related models will be able to remove the AGPL-3.0 license? 

Understanding the projected timeline would be very helpful for our planning purposes.

**Reply by KevinHuSh** [reply]
Already fixed.

**Reply by dpnotnull** [reply]
I git clone https://github.com/infiniflow/ragflow yesterday, and run below deepdoc demo script:

python deepdoc/vision/t_ocr.py --inputs XXXX --output_dir XXXX

Then downloading the model file:
![image](https://github.com/user-attachments/assets/f641374d-8775-4321-93ed-4ed2f44ad611)

**Reply by dpnotnull** [reply]
And when I check the metadata of some onnx model file

onnx inspect -m tsr.onnx

Still display : 

...
 meta.{} = {} description Ultralytics best model trained on PubTables-1M-yolo-datasets/data.yaml
  meta.{} = {} author Ultralytics
  meta.{} = {} license AGPL-3.0 https://ultralytics.com/license
  meta.{} = {} date 2024-03-25T04:36:35.872713
  meta.{} = {} version 8.1.18
  meta.{} = {} stride 32
  meta.{} = {} task detec
...

**Reply by dpnotnull** [reply]
Did I misconfigure any relevant environment variables, script parameters, or overlook any other detailed settings?

**Reply by KevinHuSh** [reply, marked as answer]
You could download from huggingface.
https://huggingface.co/InfiniFlow/deepdoc

**Reply by dpnotnull** [reply]
It's apache-2.0 now. 

meta.{} = {} license Apache-2.0 http://www.apache.org/licenses/LICENSE-2.0

Many thanks~ 🙏🙏🙏
